query.py

```diff
@@ -53,25 +56,44 @@ class QueryOr(Query):
     def __init__(self, ops):
         super().__init__()
         self.ops = ops
+        self.ops = self._flatten_ops()
         self.union = None
         self.size = None
 
-    def evaluate(self, inverted_index):
+    def _flatten_ops(self):
+        ops = []
+        for op in self.ops:
+            if isinstance(op, QueryOr):
+                for _op in op._flatten_ops():
+                    ops.append(_op)
+            else:
+                ops.append(op)
+
+        return ops
+
+    def evaluate(self, inverted_index, **kwargs):
```

Comment:
I think we can remove one of the functions **_evaluate()** and the field **self.union**.

    def evaluate(self, inverted_index, forced=False):
        ops = [set(op.evaluate(inverted_index, forced=True)) for op in self.ops]
        union = ops[0]
        for op in ops[1:]:
            union.update(op)

        union = sorted(list(union))
        self.size = len(union)

        return union


For the get_size() function, it can be modified to
`def get_size(self):
        return self.size`